### PR TITLE
#76 Added a topic describing how to use extensions in the specification.

### PR DESCRIPTION
**Pull request by brynrhodes**

### CODE REVIEW

**Comment by isaacvetter** [comment]
Hey @brynrhodes,

1) I think that using an extension in the service discovery is a lot more abstract than in the CDS request (EHR sending additional data that doesn't fit elsewhere) or the CDS response (CDS Service replying with additional data. Maybe something like the below would be more useful?

```json
{
   "hookInstance" : "d1577c69-dfbe-44ad-ba6d-3e05e953b2ea",
   "fhirServer" : "http://hooks.smarthealthit.org:9080",
   "hook" : "patient-view",
   "user" : "Practitioner/example",
   "patient" : "1288992",
   "extension" : {
      "com.example.timestamp": "Monday, 27-Nov-17 22:13:25 UTC",
      "myextension-practitionerspecialty" : " "gastroenterology"
   }
}
```

Do you disagree?

**Comment by brynrhodes** [comment]
@isaacvetter , I agree and added your example (though I changed the date/time format to ISO8601).

File: docs/specification/1.0.md [hunk 1]
```diff
@@ -496,3 +496,45 @@ Access-Control-Allow-Headers | Authorization, Content-Type
...
+    {
+      "title": "Example CDS Service Discovery",
+      "hook": "patient-view",
+      "id": "patientview",
+      "prefetch": {
+        "patient": "Patient/{{Patient.id}}"
+      },
+      "description": "clinical decision support for patient view",
+      "extension": {
+          "example-client-conformance": "http://my.hooks.org/fhir/102/Conformance/patientview"
```
Comment (kpshek) [inline docs/specification/1.0.md]: For the example URL, let's use hooks.example.org as the domain ([reasoning](https://www.iana.org/domains/reserved)). This prevents us from linking to a potentially malicious/unexpected website 😄

File: docs/specification/1.0.md [hunk 2]
```diff
@@ -496,3 +496,45 @@ Access-Control-Allow-Headers | Authorization, Content-Type
...
+## Extensions
+
+The specification is not prescriptive about support for extensions. However, to support extensions, the specification reserves the name `extension` and will never define an element with that name, allowing implementations to use it to provide custom behavior and information. The value of an extension element should be a pre-coordinated JSON object.
+
+For example, an extension on a request could look like this:
+
+```json
+{
+   "hookInstance" : "d1577c69-dfbe-44ad-ba6d-3e05e953b2ea",
+   "fhirServer" : "http://hooks.smarthealthit.org:9080",
```
Comment (kpshek) [inline docs/specification/1.0.md]: For the `fhirServer`, let's use fhir.example.org as the domain (for the same reasons as below).

File: docs/specification/1.0.md (hunk 1 above)
Comment (brynrhodes) [reply docs/specification/1.0.md]: Agreed and committed.

**Comment by kpshek** [comment]
Thanks @brynrhodes - looks good!

**Pull request merged**